### PR TITLE
[opt](nereids)expr normalize after filter pushdown

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
tpch q19 improved

tpcds

before
query13 11756   3049    2930    2930
query48 1129    918     874     874
query85 1944    668     616     616
Total cold run time: 14829 ms
Total hot run time: 4420 ms
Finish tpcds queries.

after
query13 4992    3031    2940    2940
query48 1012    771     754     754
query85 795     586     536     536

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java [hunk 1]
```diff
@@ -218,6 +218,7 @@ public class Rewriter extends AbstractBatchJobExecutor {
                     // but top-down traverse can not cover this case in one iteration, so bottom-up is more
                     // efficient because it can find the new plans and apply transform wherever it is
                     bottomUp(RuleSet.PUSH_DOWN_FILTERS),
+                    topDown(new ExpressionOptimization()),
```
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java:225]: done

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by keanji-x — APPROVED** [review]

**Comment by englefly** [comment]
run p0

**Comment by englefly** [comment]
run pipelinex_p0

**Review by seawinde — APPROVED** [review]

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run pipelinex

**Comment by englefly** [comment]
run p0

**Comment by englefly** [comment]
run buildall

**Pull request merged**